src/main/java/seedu/address/model/room/Room.java

```diff
@@ -0,0 +1,125 @@
+package seedu.address.model.room;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.person.UniqueGuestList;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a room in the hotel.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public abstract class Room {
+
+    // Identity fields
+    protected final RoomNumber roomNumber;
+
+    // Data fields
+    protected final Capacity capacity;
+    protected final UniqueGuestList occupants;
+    protected final Expenses expenses;
+    protected final ReservationDates reservationDates;
+    protected final Set<Tag> tags = new HashSet<>();
+
+    public Room(RoomNumber roomNumber, Capacity capacity, UniqueGuestList occupants, Expenses expenses,
+                ReservationDates reservationDates) {
```

Comment:
When integrating with `RoomList`, would have to be careful. I actually do not foresee using this constructor.

Instead, create setters for `UniqueGuestList`, `Expenses` and `ReservationDates`. Have to depend on `RoomList` implementation.

Reply:
I disagree. The `Room` constructor is used for constructing the rooms in the hotel. This will be needed for initially populating an empty hotel with rooms. After that, it can be used for resetting of all of the hotel's data, possibly upon the `clear` command.

Reply:
"Instead, create setters for UniqueGuestList, Expenses and ReservationDates. Have to depend on RoomList implementation." 

Please elaborate.